packages/gatsby-plugin-page-creator/src/extract-query.ts

```diff
@@ -40,17 +35,18 @@ export function reverseLookupParams(
   }
 
   absolutePath.split(path.sep).forEach(part => {
-    const extracted = compose(
-      removeFileExtension,
-      extractFieldWithoutUnion
-    )(part)
-
-    const results = _.get(
-      queryResults.nodes ? queryResults.nodes[0] : queryResults,
-      // replace __ with accessors '.'
-      switchToPeriodDelimiters(extracted)
-    )
-    reversedParams[extracted] = results
+    const extracted = extractFieldWithoutUnion(part)
```

Comment:
The regex/function already supports getting the data with or without a file extension, so removing it is unnecessary.